### PR TITLE
Fetch api

### PR DESCRIPTION
**Pull request by Alejandroq12**
### Implemented Project Requirements

- **Utilized the correct React hooks**: Implemented `useEffect` and `useState` hooks for fetching data from the API and managing component state.
- **Connected to API within a functional component**: Created a new functional component, `QuoteDisplay`, to fetch and display quotes from the API Ninjas Quotes API.
- **Displayed different UI outputs based on component state**: Showed a loading state when fetching data, an error state when an error occurs, and the fetched quote when successful.
- **Created a new component for displaying quotes**: Developed the `QuoteDisplay` component to fetch and display quotes alongside the existing `Calculator` component.
- **Imported the new component into the main component**: Imported `QuoteDisplay` into `App.js` to display it on the same page as the `Calculator`.
- **Styled the quote display**: Added extra CSS to make the quote display look good.

Please let me know if you have any questions or require any changes. Thank you!

To run this project:

```
npm install
```

```
npm start
```

### CODE REVIEW

File: src/components/QuoteDisplay.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import React, { useState, useEffect } from 'react';
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (Tufahel) [inline src/components/QuoteDisplay.js:2]: - [ ] Kindly remove unnecessary comments. To fix the linters, I recommend you do not use these kinds of comments, rather try to solve it manually. You can consider this [link](https://stackoverflow.com/questions/43989739/why-and-how-can-i-fix-eslint-import-no-extraneous-dependencies-failures-on-ins) or any relevant article to fix this issue. As sometimes linter errors happen due to syntax error so, by fixing it manually can help to learn a new concept.

**Review by Tufahel — CHANGES_REQUESTED** [review]
# Status: Required Changes :recycle:
Hi @Alejandroq12 :wave: 

Well done!!! 👍 
You still need to fix some issues before moving to the next project. :running_man: 

## Highlights 🌟

Alt Text

✔️ Linters checks are passing.
✔️ You have used the correct git-flow.
✔️ Readme is professional.
✔️ Professional commit messages.

## Required Changes ♻️

> Kindly check the comments below.

## Optional suggestions 🤷‍♂️

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.
- You can mail me at [Mail](mailto:tufahel96@gmail.com) or ask any question at Slack, `Name: Tufahel`

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/QuoteDisplay.js (hunk 1 above)
Comment (Alejandroq12) [reply src/components/QuoteDisplay.js:2]: Thank you very much for your feedback.

**Comment by Alejandroq12** [comment]
Hello, dear reviewer.

While working on the project, I encountered an issue with the ESLint rule import/no-extraneous-dependencies. As per the initial configuration, importing devDependencies in the src folder was not allowed. However, I needed to import the axios library, which is a necessary dependency for this specific component.

the previous reviewer suggested a link with a solution to update the .eslintrc.json configuration to allow importing devDependencies in the src folder:

```json
"import/no-extraneous-dependencies": ["error", {
  "devDependencies": ["**/*.test.js", "**/*.spec.js", "**/setupTests.js", "src/**/*"],
  "optionalDependencies": false,
  "peerDependencies": false
}]
```
I want to emphasize that I initially did not want to fix the issue this way, considering the guidelines about linters configuration. However, the previous reviewer pointed me to a link that offered this specific solution, as he believed it was the best approach. The suggested modification allowed me to proceed with the project without violating any style guidelines and maintaining the consistency of the codebase.

I have made sure that this change does not impact any other part of the project or violate any other style guidelines. This modification has been carefully considered and is crucial for the proper functioning of the component, as suggested by the previous reviewer.

I kindly request you to consider this explanation and allow this specific change in the linter configuration for the successful completion of the project, keeping in mind that it was a solution provided by a previous reviewer.

Thank you for your understanding.

Best regards,
Note: by saying this I am very thankful with the previous reviewer for giving me this option but at the same time I am not sure if it was the best approach. Could you please tell me something about that? Thank you very much.

**Review by AmaduKamara — APPROVED** [review]
# Status: Approved 🟢 🟢 🟢 

Hi @Alejandroq12,

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! 🎉 

## Highlights:

✅ You used Gitflow correctly 👍🏽
✅ The Linter checks passed successfully 👍🏽
✅ Used the `useEffect` hook correctly 👍🏽
✅ Quotes are fetched correctly and displayed on the screen as expected 👍🏽

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**